File: csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private List<int> _scores;

    public HighScores(List<int> list) => _scores = list;

    public List<int> Scores() => _scores;

    public int Latest() => _scores[_scores.Count-1];

    public int PersonalBest() => _scores.Max();

    public List<int> PersonalTopThree() => _scores.OrderByDescending(o => o).Take(3).ToList<int>();
}

File: ruby/high-scores/high_scores.rb
class HighScores

    def initialize(scoreList)
        @scoreList=scoreList
    end

    def scores()
        return @scoreList
    end

    def personal_best()
        return @scoreList.max()
    end

    def latest()
        return @scoreList.last()
    end

    def personal_top_three()
        return @scoreList.sort { |x,y| y <=> x }.take(3)
    end

    def latest_is_personal_best?
        return self.latest() == self.personal_best()
    end
end